src/lib/util/Config.ts

```diff
@@ -88,31 +117,48 @@ export class Config {
 		return !e || e.disabled;
 	}
 
-	public getEngineConfig(name: string): EngineConfigContent {
-		return this.configContent.engines.find(e => e.name === name);
+	public async getSupportedLanguages(engine: ENGINE): Promise<string[]> {
+		const value = await this.getConfigValue('supportedLanguages', engine, this.typeChecker.stringArrayCheck);
+		return value as Array<string>;
+	}
+
+	public async getTargetPatterns(engine: ENGINE): Promise<string[]> {
+		const value = await this.getConfigValue('targetPatterns', engine, this.typeChecker.stringArrayCheck);
+		return value as Array<string>;
 	}
 
-	public async getSupportedLanguages(engineName: string): Promise<string[]> {
-		const ecc = this.getEngineConfig(engineName);
-		// If the config specifies supported languages, use those.
-		if (ecc.supportedLanguages && ecc.supportedLanguages.length > 0) {
-			this.logger.trace(`Retrieving supported languages ${ecc.supportedLanguages} from engine ${engineName}`);
-			return ecc.supportedLanguages;
+	private async getConfigValue(propertyName: string, engine: ENGINE, typeChecker: (any, string, ENGINE) => boolean): Promise<string[]> {
```

Comment:
This is a slightly modified version of Josh's code for getSupportedLanguages(). The additional check here is to see if config block for the given engine exists and use a default if it doesn't.